src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -430,9 +431,16 @@ export class DashboardAppController {
               .getStateTransfer(scopedHistory())
               .getIncomingEmbeddablePackage();
             if (incomingState) {
-              container.addNewEmbeddable<SavedObjectEmbeddableInput>(incomingState.type, {
-                savedObjectId: incomingState.id,
-              });
+              if ('id' in incomingState) {
```

Comment:
nit: I'd put these type guards near `EmbeddablePackageState`

Reply:
Could you elaborate a bit on this? Typescript complains once we do `addNewEmbeddable` as it doesn't know if there's `id` field in that object (similarly for `input`). I don't know of another way of doing this.